tests/webTables.spec.ts

```diff
@@ -0,0 +1,515 @@
+import { spec } from 'node:test/reporters';
+
+const timeout = 7000
+
+test.beforeEach(async ({ page }) => {
+    await page.goto('/')
+
+});
+
+test('Validate the pet name city of the owner', async ({ page }) => {
+    /* 1. Select the OWNERS menu item in the navigation bar and then select "Search" from the drop-down menu */
+    // Navigate to the owners table
+    await page.getByText('Owners').click()
+    await page.getByText('Search').click()
+
+    /* 2. In the list of Owners, locate the owner by the name "Jeff Black". 
+    Add the assertions that this owner is from the city of "Monona" and he has a pet with a name "Lucky" */
+    // Find Jeff Black in the table by searching
+    await page.getByRole('textbox').fill("black")
+    await page.getByText('Find Owner').click()
+
+    const row = page.getByRole('row').locator('td')
+    await expect(row.first()).toHaveText('Jeff Black')
+
+    // Validate that Jeff is from Monona
+    await expect(row.nth(2)).toHaveText('Monona')
+
+    // Validate that Jeff has a pet named Lucky
+    await expect(row.nth(4)).toContainText('Lucky')
+
+});
+
+test('Validate owners count of the Madison city', async ({ page }) => {
+    /* 1. Select the OWNERS menu item in the navigation bar and then select "Search" from the drop-down menu */
+    // Navigate to the owners table
+    await page.getByText('Owners').click()
+    await page.getByText('Search').click()
+
+    /* 2. In the list of Owners, locate all owners who live in the city of "Madison". 
+    Add the assertion that the total number of owners should be 4 */
+    // Wait for all table contents to load
+    await page.waitForTimeout(timeout)
+    const allRows = await page.getByRole('row').all()
+    let counter = 0
+
+    // Iterate through each row of the table
+    for (const row of allRows) {
+        // Collect each cell in the row
+        const rowData = row.getByRole('cell')
+
+        // Make sure the row has cells
+        if((await rowData.all()).length > 0){
+            if (await rowData.nth(2).textContent() === 'Madison') {
+                // Count the occurrence of the city of Madison
+                counter += 1
+            }
+        }
+    }
+    // Validate that there are only 4 'Madison' entries in the table
+    expect(counter).toEqual(4)
+});
+
+test('Validate search by Last Name', async ({ page }) => {
+    /* 1. Select the OWNERS menu item in the navigation bar and then select "Search" from the drop-down menu */
+    // Navigate to the owners table
+    await page.getByText('Owners').click()
+    await page.getByText('Search').click()
+
+    /* 2. On the Owners page, in the "Last name" input field, type the last name "Black" and click the  "Find Owner" button */
+    const searchField = page.getByRole('textbox')
+    const searchButton = page.getByText('Find Owner')
+
+    // Search for Jeff Black
+    await searchField.fill("Black")
+    await searchButton.click()
+
+    await page.waitForTimeout(timeout)
```

Comment:
I had so much trouble with things on my end that I just did that.  Maybe it was because of my approach to this problem set.  I'll remove them happily.

Reply:
I think that this might have just been my approach, because I no longer need these.